lib/ui/painting.dart

```diff
@@ -896,25 +896,35 @@ abstract class Codec extends NativeFieldWrapperClass2 {
   /// Fetches the next animation frame.
   ///
   /// Wraps back to the first frame after returning the last frame.
-  ///
+  Future<FrameInfo> getNextFrame() async {
+    return _futurize(_getNextFrame);
+  }
+
   /// Returns an error message on failure, null on success.
-  String getNextFrame(NextFrameInfoCallback callback) native "Codec_getNextFrame";
+  String _getNextFrame(NextFrameInfoCallback callback) native "Codec_getNextFrame";
 
   /// Release the resources used by this object. The object is no longer usable
   /// after this method is called.
   void dispose() native "Codec_dispose";
 }
 
-/// Callback signature for [imageCodecFromList].
+/// Callback signature for [_instantiateImageCodec].
```

Comment:
If it's a signature only for a private class, the typedef itself should also be private.

Reply:
changed to use  _Callback<Codec> instead